Jenkinsfile.windows

```diff
@@ -0,0 +1,281 @@
+    booleanParam(name: 'RSTUDIO_SKIP_QT', defaultValue: false, description: 'Skips installing and bulding for QT')
+    booleanParam(name: 'DAILY', defaultValue: false, description: 'Runs daily build if true')
+    booleanParam(name: 'PUBLISH', defaultValue: true, description: 'Runs publish stage if true')
+    gitParameter defaultValue: "${env.GIT_BRANCH}", name: 'COMMIT_HASH', type: 'PT_REVISION'
+  }
+
+  environment {
+    PACKAGE_OS = 'Windows'
+    AWS_ACCOUNT_ID = '749683154838'
+  }
+
+  stages {
+
+    stage('Load Utils') {
+      steps {
+        script {
+          sh 'printenv'
+          sh "echo 'Loading utils from ${env.WORKSPACE}/utils.groovy'"
+          utils = load "${env.WORKSPACE}/utils.groovy"
+        }
+      }
+    }
+
+    stage('Versioning') {
+      steps {
+        script {
+          (RSTUDIO_VERSION,
+            RSTUDIO_VERSION_MAJOR,
+            RSTUDIO_VERSION_MINOR,
+            RSTUDIO_VERSION_PATCH,
+            RSTUDIO_VERSION_SUFFIX) = utils.getVersion()
+          RSTUDIO_RELEASE = "${RSTUDIO_VERSION_MAJOR}.${RSTUDIO_VERSION_MINOR}.${RSTUDIO_VERSION_PATCH}${RSTUDIO_VERSION_SUFFIX}"
+          RSTUDIO_FLOWER = readFile(file: 'version/RELEASE').replaceAll(" ", "-").toLowerCase().trim()
+          currentBuild.displayName = "${RSTUDIO_RELEASE}"
+          IS_PRO = RSTUDIO_VERSION_SUFFIX.contains('pro')
+        }
+      }
+    }
+
+    stage('Use Windows Agent') {
+      agent { label "windows" }
+
+      stages {
+        stage('Prepare Windows Container') {
+          stages {
+            stage('Checkout') {
+              when { expression { params.COMMIT_HASH !=  env.GIT_BRANCH } }
+              steps {
+                echo "Commit_hash value: ${params.COMMIT_HASH}"
+                checkout([$class: 'GitSCM',
+                   branches: [[name: "${params.COMMIT_HASH}"]],
+                   extensions: [],
+                   userRemoteConfigs: [[credentialsId: 'github-rstudio-jenkins', url: 'https://github.com/rstudio/rstudio']]])
+              }
+            }
+            stage('Pull-Build-Push') {
+              steps {
+                withCredentials([usernameColonPassword(credentialsId: 'github-rstudio-jenkins', variable: "github_login")]) {
+                  pullBuildPush(
+                    image_name: 'jenkins/ide',
+                    dockerfile: "docker/jenkins/Dockerfile.windows",
+                    image_tag: "windows-${RSTUDIO_FLOWER}",
+                    build_args:  "--build-arg GITHUB_LOGIN=${github_login}",
+                    build_arg_jenkins_uid: null, // Ensure linux-only step is not run on windows (id -u jenkins)
+                    build_arg_jenkins_gid: null, // Ensure linux-only step is not run on windows (id -g jenkins)
+                    build_arg_docker_gid: null, // Ensure linux-only step is not run on windows (stat -c %g /var/run/docker.sock)
+                    retry_image_pull: 5)
+                }
+              }
```

Comment:
If we'd like, we can move this into the [docker image building job](https://github.com/rstudio/rstudio/pull/12606)

Reply:
Yeah let's consolidate the image builds. I'll remove this stage.